internal/pkg/aws/cloudwatchlogs/cloudwatchlogs.go

```diff
@@ -78,98 +60,110 @@ func New(s *session.Session) *CloudWatchLogs {
 
 // LogEvents returns an array of Cloudwatch Logs events.
-func (c *CloudWatchLogs) LogEvents(logGroupName string, streamLastEventTime map[string]int64, opts ...GetLogEventsOpts) (*LogEventsOutput, error) {
+func (c *CloudWatchLogs) LogEvents(opts *LogEventsOpts) (*LogEventsOutput, error) {
 	var events []*Event
-	var in *cloudwatchlogs.GetLogEventsInput
-	logStreamNames, err := c.logStreams(logGroupName)
+	// Set default value
+	in := defaultGetLogEventsInput(opts)
+	logStreams, err := c.logStreams(opts.LogGroup, opts.LogStream...)
 	if err != nil {
 		return nil, err
 	}
-	for _, logStreamName := range logStreamNames {
-		in = &cloudwatchlogs.GetLogEventsInput{
-			LogGroupName:  aws.String(logGroupName),
-			LogStreamName: logStreamName,
-			Limit:         aws.Int64(10), // default to be 10
-		}
-		for _, opt := range opts {
-			opt(in)
-		}
-		if streamLastEventTime[*logStreamName] != 0 {
+	if opts.StreamLastEventTime == nil {
+		opts.StreamLastEventTime = make(map[string]int64)
+	}
+	for _, logStream := range logStreams {
+		// Set override value
+		in.SetLogStreamName(logStream)
+		if opts.StreamLastEventTime[logStream] != 0 {
 			// If last event for this log stream exists, increment last log event timestamp
 			// by one to get logs after the last event.
-			in.SetStartTime(streamLastEventTime[*logStreamName] + 1)
+			in.SetStartTime(opts.StreamLastEventTime[logStream] + 1)
 		}
 		// TODO: https://github.com/aws/copilot-cli/pull/628#discussion_r374291068 and https://github.com/aws/copilot-cli/pull/628#discussion_r374294362
 		resp, err := c.client.GetLogEvents(in)
 		if err != nil {
-			return nil, fmt.Errorf("get log events of %s/%s: %w", logGroupName, *logStreamName, err)
+			return nil, fmt.Errorf("get log events of %s/%s: %w", opts.LogGroup, logStream, err)
 		}
 
 		for _, event := range resp.Events {
 			log := &Event{
-				LogStreamName: trimLogStreamName(*logStreamName),
+				LogStreamName: logStream,
 				IngestionTime: aws.Int64Value(event.IngestionTime),
 				Message:       aws.StringValue(event.Message),
 				Timestamp:     aws.Int64Value(event.Timestamp),
 			}
 			events = append(events, log)
 		}
 		if len(resp.Events) != 0 {
-			streamLastEventTime[*logStreamName] = *resp.Events[len(resp.Events)-1].Timestamp
+			opts.StreamLastEventTime[logStream] = *resp.Events[len(resp.Events)-1].Timestamp
 		}
 	}
 	sort.SliceStable(events, func(i, j int) bool { return events[i].Timestamp < events[j].Timestamp })
-	var truncatedEvents []*Event
-	if len(events) >= int(*in.Limit) {
-		truncatedEvents = events[len(events)-int(*in.Limit):]
-	} else {
-		truncatedEvents = events
+	limit := int(aws.Int64Value(in.Limit))
+	if limit != 0 {
+		return &LogEventsOutput{
+			Events:        truncateEvents(limit, events),
+			LastEventTime: opts.StreamLastEventTime,
+		}, nil
 	}
 	return &LogEventsOutput{
-		Events:        truncatedEvents,
-		LastEventTime: streamLastEventTime,
+		Events:        events,
+		LastEventTime: opts.StreamLastEventTime,
```

Comment:
nit: to be consistent and obvious these two things should be matched
```suggestion
		StreamLastEventTime: opts.StreamLastEventTime,
```